openmetadata-service/src/main/java/org/openmetadata/service/resources/datamodels/DataModelResource.java

```diff
@@ -0,0 +1,443 @@
+  }
+
+  @PATCH
+  @Path("/{id}")
+  @Operation(
+      operationId = "patchDataModel",
+      summary = "Update a dataModel",
+      tags = "dataModels",
+      description = "Update an existing dataModel using JsonPatch.",
+      externalDocs = @ExternalDocumentation(description = "JsonPatch RFC", url = "https://tools.ietf.org/html/rfc6902"))
+  @Consumes(MediaType.APPLICATION_JSON_PATCH_JSON)
+  public Response patch(
+      @Context UriInfo uriInfo,
+      @Context SecurityContext securityContext,
+      @Parameter(description = "Id of the dataModel", schema = @Schema(type = "UUID")) @PathParam("id") UUID id,
+      @RequestBody(
+              description = "JsonPatch with array of operations",
+              content =
+                  @Content(
+                      mediaType = MediaType.APPLICATION_JSON_PATCH_JSON,
+                      examples = {
+                        @ExampleObject("[" + "{op:remove, path:/a}," + "{op:add, path: /b, value: val}" + "]")
+                      }))
+          JsonPatch patch)
+      throws IOException {
+    return patchInternal(uriInfo, securityContext, id, patch);
+  }
+
+  @PUT
+  @Operation(
+      operationId = "createOrUpdateDataModel",
+      summary = "Create or update dataModel",
+      tags = "dataModels",
+      description = "Create a dataModel, it it does not exist or update an existing dataModel.",
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "The updated dataModel ",
+            content = @Content(mediaType = "application/json", schema = @Schema(implementation = DataModel.class)))
+      })
+  public Response createOrUpdate(
+      @Context UriInfo uriInfo, @Context SecurityContext securityContext, @Valid CreateDataModel create)
+      throws IOException {
+    DataModel dataModel = getDataModel(create, securityContext.getUserPrincipal().getName());
+    return createOrUpdate(uriInfo, securityContext, dataModel);
+  }
+
+  @PUT
+  @Path("/{id}/followers")
+  @Operation(
+      operationId = "addFollowerToDataModel",
+      summary = "Add a follower",
+      tags = "dataModels",
+      description = "Add a user identified by `userId` as followed of this data model",
+      responses = {
+        @ApiResponse(responseCode = "200", description = "OK"),
+        @ApiResponse(responseCode = "404", description = "DataModel for instance {id} is not found")
+      })
+  public Response addFollower(
+      @Context UriInfo uriInfo,
+      @Context SecurityContext securityContext,
+      @Parameter(description = "Id of the data model", schema = @Schema(type = "UUID")) @PathParam("id") UUID id,
+      @Parameter(description = "Id of the user to be added as follower", schema = @Schema(type = "UUID")) UUID userId)
+      throws IOException {
+    return dao.addFollower(securityContext.getUserPrincipal().getName(), id, userId).toResponse();
+  }
+
+  @DELETE
+  @Path("/{id}/followers/{userId}")
```

Comment:
Are we going to allow separate followers for dataModel or would we just allow following dashboards? cc @harshach

Reply:
We are allowing separate followers for Charts, and they are part of Dashboards, I followed the same approach.